source/solvers/tracer_assemblers.cc

```diff
@@ -241,6 +241,102 @@ TracerAssemblerCore<dim>::assemble_rhs(TracerScratchData<dim>    &scratch_data,
 template class TracerAssemblerCore<2>;
 template class TracerAssemblerCore<3>;
 
+template <int dim>
+void
+TracerAssemblerDGCore<dim>::assemble_matrix(
+  TracerScratchData<dim>    &scratch_data,
+  StabilizedMethodsCopyData &copy_data)
+{
+  // Scheme and physical properties
+  const std::vector<double> &diffusivity_vector =
+    scratch_data.tracer_diffusivity;
+
+  // Loop and quadrature informations
+  const auto        &JxW_vec    = scratch_data.JxW;
+  const unsigned int n_q_points = scratch_data.n_q_points;
+  const unsigned int n_dofs     = scratch_data.n_dofs;
+
+  // Copy data elements
+  auto &local_matrix = copy_data.local_matrix;
+
+  // assembling local matrix and right hand side
```

Comment:
```suggestion
  // Assembling local matrix
```

Reply:
I mostly mean pre-allocating the memory for the arrays instead of allocating them on the fly. When we re-allocate for every cell, we do pay a price for that. That will be done in a follow-up PR.

Reply:
I see!